### DISCUSSION TITLE
Issue with Mitsuba-Blender Plugin Unregistering

### DISCUSSION THREAD

**Discussion by YanQiSun** [mitsuba-renderer/mitsuba-blender#91] [Q&A]
When I try to enable the Mitsuba-Blender plugin in Blender, I encounter the following error:

Copy code
```
Error: Traceback (most recent call last):
  File "/home/sun/blender/blender-3.3.1-linux-x64/3.3/scripts/modules/addon_utils.py", line 295, in enable
    mod.unregister()
  File "/home/sun/blender/blender-3.3.1-linux-x64/3.3/scripts/addons/mitsuba-blender/__init__.py", line 307, in unregister
    unregister_class(cls)
RuntimeError: unregister_class(...):, missing bl_rna attribute from 'RNAMeta' instance (may not be registered)
```

This issue seems to be related to the plugin's unregistering process, specifically, the 'RNAMeta' instance lacking the 'bl_rna' attribute, which may result in an unregistered state. I've attempted to reload the plugin and check its relevant settings, but the problem persists. I would like to inquire if there's any solution to address this issue.My Blender version is 3.3.1,  Mitsuba is 3.5.

**Comment by bathal1** [comment]
I cannot reproduce this issue on my end. Have you installed the add-on using the ZIP file in the latest [release](https://github.com/mitsuba-renderer/mitsuba-blender/releases/tag/v0.3.0)?

Does the issue happen when you tick the box in the add-ons list to enable this one ?

Also, which version of Mitsuba are you referencing here ? The correct one should be installed by the add-on itself.

**Reply by YanQiSun** [reply]
Thank you for your response. I've confirmed that I installed the add-on using the ZIP file from the latest release. I've encountered an issue when attempting to automate the activation of the Mitsuba-Blender plugin through Python code. While I can successfully enable and use the plugin within Blender's interface, I encounter the 'missing bl_rna attribute' error when trying to automate the process.

I'm currently using Mitsuba version 3.5.

I've also tried programmatically enabling the add-on using the following Python code:
import addon_utils
addon_name = "mitsuba-blender"
addon_utils.enable(addon_name)

However, the error occurs at the line where I attempt to enable the add-on through code. I'm wondering if there's a specific procedure or consideration I should take into account when activating the plugin programmatically. Do you have any insights or recommendations for handling the plugin activation through Python?

I appreciate your continued support in resolving this issue.
